ztp/siteconfig-generator/siteConfig/siteConfigBuilder.go

```diff
@@ -369,87 +373,162 @@ func (scbuilder *SiteConfigBuilder) getWorkloadManifest(cpuSet string, role stri
+				manifestFile, err := ReadExtraManifestResourceFile(filePath)
+				if err != nil {
+					return dataMap, doNotMerge, err
+				}
 
-			manifestFileStr, err := addZTPAnnotationToManifest(string(manifestFile))
-			if err != nil {
-				return dataMap, err
+				manifestFileStr, err := addZTPAnnotationToManifest(string(manifestFile))
+				if err != nil {
+					return dataMap, doNotMerge, err
+				}
+				dataMap[file.Name()] = manifestFileStr
+
+				// assuming user provided custom CRs are not in the first directory path
+				// for a single directory path, only CRs under workload will be
+				// included in doNotMerge
+				// if len(dirFilesArray) > 1 && i != 0 {
+				// 	// user provided CRs don't need to be merged
+				// 	doNotMerge[file.Name()] = true
+				// }
 			}
-			dataMap[file.Name()] = manifestFileStr
+
 		}
 	}
 
 	// Adding workload partitions MC only for SNO clusters.
+	// we have to find out which dir contains workload dir
 	if clusterSpec.ClusterType == SNO {
 		for node := range clusterSpec.Nodes {
 			cpuSet := clusterSpec.Nodes[node].Cpuset
 			role := clusterSpec.Nodes[node].Role
 			if cpuSet != "" {
-				k, v, err := scbuilder.getWorkloadManifest(cpuSet, role)
-				if err != nil {
-					errStr := fmt.Sprintf("Error could not read WorkloadManifest %s %s\n", clusterSpec.ClusterName, err)
-					return dataMap, errors.New(errStr)
-				} else {
-					data, err := addZTPAnnotationToManifest(v.(string))
-					if err != nil {
-						return dataMap, err
+				//
+				for _, v := range dirFilesArray {
+					for _, file := range v.Files {
+						if file.Name() == workloadPath {
+							k, v, err := scbuilder.getWorkloadManifest(v.Directory, cpuSet, role)
+							if err != nil {
+								errStr := fmt.Sprintf("Error could not read WorkloadManifest %s %s\n", clusterSpec.ClusterName, err)
+								return dataMap, doNotMerge, errors.New(errStr)
+							} else {
+								data, err := addZTPAnnotationToManifest(v.(string))
+								if err != nil {
+									return dataMap, doNotMerge, err
+								}
+								dataMap[k] = data
+								// Exclude the workload manifest
+								doNotMerge[k] = true
+							}
+						}
+
 					}
-					dataMap[k] = data
-					// Exclude the workload manifest
-					doNotMerge[k] = true
 				}
 			}
 		}
 	}
 
+	return dataMap, doNotMerge, err
+}
+
+func (scbuilder *SiteConfigBuilder) getExtraManifest(dataMap map[string]interface{}, clusterSpec Clusters) (map[string]interface{}, error) {
+	// Figure out the list of node roles we need to support in this cluster
+	roles := map[string]bool{}
+	doNotMerge := map[string]bool{}
+	var err error
+
+	for _, node := range clusterSpec.Nodes {
+		roles[node.Role] = true
+	}
+
+	if clusterSpec.ExtraManifests.SearchPaths != nil {
+		dataMap, doNotMerge, err = scbuilder.getExtraManifestMaps(clusterSpec.ExtraManifests.SearchPaths, roles, clusterSpec)
+		if err != nil {
+			return dataMap, err
+		}
+	} else {
```

Comment:
if `clusterSpec.ExtraManifests.SearchPaths` is not defined, the value is nil, but  `scbuilder.scBuilderExtraManifestPath` is never nil because it is hard coded container path. So, here the logic is to fall back to container path for getting extra manifests if `SearchPaths` were nil.